packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2928,7 +2928,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       widget.controller.addListener(_didChangeTextEditingValue);
       _updateRemoteEditingValueIfNeeded();
     }
-    if (widget.controller.selection != oldWidget.controller.selection) {
+
+    if (_selectionOverlay != null
+        && (widget.contextMenuBuilder != oldWidget.contextMenuBuilder
+            || widget.selectionControls != oldWidget.selectionControls
+            || widget.onSelectionHandleTapped != oldWidget.onSelectionHandleTapped
+            || widget.dragStartBehavior != oldWidget.dragStartBehavior
+            || widget.magnifierConfiguration != oldWidget.magnifierConfiguration)) {
+      final bool shouldShowToolbar = _selectionOverlay!.toolbarIsVisible;
+      final bool shouldShowHandles = _selectionOverlay!.handlesVisible;
+      _selectionOverlay!.dispose();
+      _selectionOverlay = _createSelectionOverlay();
+      if (shouldShowToolbar || shouldShowHandles) {
+        SchedulerBinding.instance.addPostFrameCallback((Duration _) {
+          if (shouldShowToolbar) {
+            _selectionOverlay!.showToolbar();
+          }
+          if (shouldShowHandles) {
+            _selectionOverlay!.showHandles();
+          }
+        });
```

Comment:
what does this look like from an animation perspective? will it cause weird effects if someone is trying to, say, animate the `magnifierConfiguration`?

Reply:
I tried it and it does as I feared. I've created an issue to track this: https://github.com/flutter/flutter/issues/142763


https://github.com/flutter/flutter/assets/389558/eafde5e1-5f66-4f15-ac36-5960f84c2198